rfcs/20200817-jit-code-generation-for-aarch64/README.md

```diff
@@ -0,0 +1,256 @@
+for x64.
+
+```
+static const std::map<conv_impl_key_t, std::vector<pd_create_f>> impl_list_map {
+    // FWD fp
+    {{forward, f32, f32, f32}, {
+        CPU_INSTANCE_X64(jit_avx512_common_dw_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_common_1x1_convolution_fwd_f32_t)
+        CPU_INSTANCE_X64(jit_avx512_core_f32_wino_conv_2x3_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_core_f32_wino_conv_4x3_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_common_convolution_winograd_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_common_convolution_fwd_t<f32>)
+        CPU_INSTANCE_X64(jit_avx2_dw_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_avx2_1x1_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_sse41_dw_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_sse41_1x1_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_avx2_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_sse41_convolution_fwd_t)
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_dw_convolution_fwd_t)         <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_1x1_convolution_fwd_f32_t)    <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_core_f32_wino_conv_2x3_fwd_t)        <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_core_f32_wino_conv_4x3_fwd_t)        <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_convolution_winograd_fwd_t)   <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_convolution_fwd_t<f32>)       <- Added
+        CPU_INSTANCE(gemm_convolution_fwd_t)
+        CPU_INSTANCE(ref_convolution_fwd_t<f32>)
+        CPU_INSTANCE(ref_fused_convolution_fwd_t)
+        nullptr,
+    }},
+```
+
+The figure below shows the types of SIMD instructions of AArch64. 
+AArch64 has SIMD & FP(Floating point) instructions as 128-bit long SIMD 
+instructions, and SVE instructions as more than or equal to 128-bit SIMD 
+instructions.SVE and SIMD & FP instructions have relationship where SVE 
+extends SIMD & FP instructions. Bit width of SVE instructions is implementation 
+dependent. AArch64 specification allows for 128 *N* (*N* = 1, 2, ..., 16).
+
+![SIMD instructions of AArch64](aarch64_isa.png)
+
+A primitive that uses only SIMD & FP instructions includes `aarch64_simdfp` in 
+its name.
+
+  ~~~c++
+    CPU_INSTANCE_AARCH64(jit_aarch64_simdfp_*)
+  ~~~
+
+A primitive that uses SVE and SIMD &FP instructions includes `aarch64_sve_B` in 
+its name, where *B* is the assumed SVE bit-width. The underscore between SVE 
+and *B* is for the next instruction extension, namely SVE2.
+
+  ~~~c++
+    CPU_INSTANCE_AARCH64(jit_aarch64_sve_B_*)
+  ~~~
+
+## Test & CI
+
+JIT code for AArch64 can be tested in the same way as for JIT code for x64 by 
+`tests/gtest`.
+
+CI for AArch64 can be done using [QEMU](https://www.qemu.org/ "QEMU"). However, 
+QEMU takes long time for CI, we plans to provide an AArch64 architecture server 
+for CI on Internet. I'd like to make another issue or RFC to discuss this.
```

Comment:
This is just a thought or a friendly suggestion. It might be worth also adding an Arm Compute Library build to this CI (even if just to test the primitive fallback mechanism).

Reply:
We plan to prepare an Arm server for CI of all functions implemented for AArch64, off coursce ArmCL is included.